Extension/src/LanguageServer/settingsPanel.ts

```diff
@@ -0,0 +1,213 @@
+        // Notify listener config panel is not active
+        if (this.configDirty) {
+            let viewState: ViewStateEvent = { isActive: false };
+            this.settingsPanelViewStateChanged.fire(viewState);
+            this.configDirty = false;
+        }
+
+        if (this.disposablesPanel) { 
+            this.disposablesPanel.dispose(); 
+            this.panel = undefined;
+        }
+    }
+
+    private updateWebview(configuration: config.Configuration): void {
+        this.configValues = configuration;
+        // Send a message to the webview webview to update the settings from json.
+        if (this.panel) {
+           this.panel.webview.postMessage({ command: 'update', config: configuration });
+        }
+    }
+
+    private onViewStateChanged(e: vscode.WebviewPanelOnDidChangeViewStateEvent): void {
+        let viewState: ViewStateEvent = { isActive: e.webviewPanel.active };
+        if (this.configDirty || e.webviewPanel.active) {
+            this.settingsPanelViewStateChanged.fire(viewState);
+            this.configDirty = false;
+        }
+    }
+
+    private onMessageReceived(message: any): void {
+        if (message === null) {
+            return;
+        }
+        switch (message.command) {
+            case 'change':
+                this.updateConfig(message);
+        }
+    }
+
+    private updateConfig(message: any): void {
+        let entries: string[];
+        this.configDirty = true;
+
+        switch (message.key) {
+            case ElementId.ActiveConfig:
+                this.configValues.name = message.value;
+                break;
+            case ElementId.CompilerPath:
+                this.configValues.compilerPath = message.value;
+                break;
+            case ElementId.IncludePath:
+                entries = message.value.split("\n");
+                this.configValues.includePath = entries;
+                break;
+            case ElementId.Defines:
+                entries = message.value.split("\n");
+                this.configValues.defines = entries;
+                break;
+            case ElementId.IntelliSenseMode:
+                this.configValues.intelliSenseMode = message.value;
+                break;
+            case ElementId.cStandard:
+                this.configValues.cStandard = message.value;
+                break;
+            case ElementId.cppStandard:
+                this.configValues.cppStandard = message.value;
+                break;
+        }
+    }
+
+    private getHtml(): string {
+        let content: string | undefined;
+        content = fs.readFileSync(util.getExtensionFilePath("ui/settings.html")).toString();
+
+        content = content.replace(
+            /{{root}}/g, 
+            vscode.Uri.file(util.extensionContext.extensionPath)
+            .with({ scheme: 'vscode-resource' })
+            .toString());
+
+        content = content.replace(
+            /{{nonce}}/g, 
+            this.getNouce());

```

Comment:
Is it 'nonce' or 'nouce'?  I don't know what either of those means.

Reply:
it should be `nonce`, used to whitelist an inline script. Whenever a new webview is created, a random `nonce ` is generated for the html that's used by the webview.

Reply:
(dyslexic sometimes)

Reply:
I believe it should be [nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce).